data/finset.lean

```diff
@@ -992,6 +989,57 @@ sort_eq _ _
 
 @[simp] theorem sort_to_finset [decidable_eq α] (s : finset α) : (sort r s).to_finset = s :=
 list.to_finset_eq (sort_nodup r s) ▸ eq_of_veq (sort_eq r s)
+
 end sort
 
+variable [decidable_eq α]
+
+theorem disjoint_iff_inter_eq_empty [decidable_eq α] {s t : finset α} : disjoint s t ↔ s ∩ t = ∅ :=
+iff.rfl
+
+theorem disjoint_left {s t : finset α} : disjoint s t ↔ ∀ {a}, a ∈ s → a ∉ t :=
+by simp [disjoint_iff_inter_eq_empty, ext, mem_inter]
+
+theorem disjoint_right {s t : finset α} : disjoint s t ↔ ∀ {a}, a ∈ t → a ∉ s :=
+by rw [_root_.disjoint_comm, disjoint_left]
+
+theorem disjoint_iff_ne {s t : finset α} : disjoint s t ↔ ∀ a ∈ s, ∀ b ∈ t, a ≠ b :=
+by simp [disjoint_left, imp_not_comm]
+
+theorem disjoint_of_subset_left {s t u : finset α} (h : s ⊆ u) (d : disjoint u t) : disjoint s t :=
+disjoint_left.2 (λ x m₁, (disjoint_left.1 d) (h m₁))
+
+theorem disjoint_of_subset_right {s t u : finset α} (h : t ⊆ u) (d : disjoint s u) : disjoint s t :=
+disjoint_right.2 (λ x m₁, (disjoint_right.1 d) (h m₁))
+
+@[simp] theorem disjoint_empty_left (s : finset α) : disjoint ∅ s := bot_inf_eq
+
+@[simp] theorem disjoint_empty_right (s : finset α) : disjoint s ∅ := inf_bot_eq
+
+@[simp] theorem singleton_disjoint {s : finset α} {a : α} : disjoint (singleton a) s ↔ a ∉ s :=
```

Comment:
Should `singleton_disjoint` be `disjoint_singleton_left` and `disjoint_singleton` be `disjoint_singleton_right` for consistency? I'm not sure what the mathlib naming policy would say here.

Reply:
The policy has been to use positional naming, which this theorem is consistent with. I think there is a PR (yours?) proposing switching to left/right naming, and we can change this as well if that goes through. In any case that's an orthogonal issue.

Reply:
@digama0:

> The policy has been to use positional naming, which this theorem is consistent with.

I was trying to find where that policy was written down, but I can't. I don't think it's clear or consistent, either. For example, [`docs/naming.md`](https://github.com/leanprover/mathlib/blob/a97101d2df5d186848075a2d0452f6a04d8a13eb/docs/naming.md) says:

> Sometimes the word "left" or "right" is helpful to describe variants of a theorem.

And I don't mean to criticize here about consistency in the past. I understand that things have developed organically. But it is useful to strive for consistency going forward, I think.

> I think there is a PR (yours?) proposing switching to left/right naming, and we can change this as well if that goes through.

No, I posted something on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/subject/naming.20of.20binary.20properties.20with.20left.20and.20right). But I will make an GitHub issue for it so that the issue does not get lost.

> In any case that's an orthogonal issue.

Orthogonal to what? I'm not suggesting that this PR use one particular naming scheme. I'm merely suggesting it use one scheme in a way that is self-consistent.